### PR TITLE
new changes for todos

### PR DESCRIPTION
**Pull request by Liavudi**
Hi i made a new changes to the delete function please, check it out.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -157,18 +157,18 @@ def log_in():
...
 
 @app.route("/todolist", methods=['GET'])
 def get_todo():
     return um.get_todo_list()
 
 
-@app.route("/todolist/<user>", methods=['DELETE'])
-def delete_todo(user):
-    return um.delete_todo(user)
+@app.route("/todolist/<user>/<todo_id>", methods=['DELETE'])
```
Comment (Liavudi) [inline app.py]: Delete user argument from the frontend and the delete function in the backend

File: app.py [hunk 2]
```diff
@@ -157,18 +157,18 @@ def log_in():
 def create_todo():
     input_data = json.loads(request.data)
     print(input_data)
```
Comment (sahare92) [inline app.py]: remove unnecessary prints

File: app.py [hunk 3]
```diff
@@ -157,18 +157,18 @@ def log_in():
 def create_todo():
     input_data = json.loads(request.data)
     print(input_data)
-    return um.create_todo_list(user_name=input_data.get('user'), todo=input_data.get('todo'))
-
+    um.create_todo_list(user_name=input_data.get('user'), todo=input_data.get('todo'))
+    return {}
```
Comment (sahare92) [inline app.py]: 1. Whenever creating a resource - return it in the response body, like this:
{
  'data': todo_data
}

2. When creating a resource successfully - a 201 status code should be returned (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201)

File: user_management.py [hunk 4]
```diff
@@ -85,19 +86,30 @@ def list_users(self) -> dict:
             raise DBError('Failed to insert the user to the db', internal_exception=exc)
 
     def create_todo_list(self,user_name, todo):
```
Comment (sahare92) [inline user_management.py]: move to another module - todos something - choose a name

File: user_management.py [hunk 5]
```diff
@@ -85,19 +86,30 @@ def list_users(self) -> dict:
             raise DBError('Failed to insert the user to the db', internal_exception=exc)
 
     def create_todo_list(self,user_name, todo):
-        print(user_name, todo)
-        self.collection.update_one({'username': user_name}, {"$set": {"todo": todo}})
+        dict = {
+            'username': user_name,
+            'todo': todo
```
Comment (sahare92) [inline user_management.py]:
```suggestion
            'text': todo

```

File: user_management.py [hunk 6]
```diff
@@ -85,19 +86,30 @@ def list_users(self) -> dict:
             raise DBError('Failed to insert the user to the db', internal_exception=exc)
 
     def create_todo_list(self,user_name, todo):
-        print(user_name, todo)
-        self.collection.update_one({'username': user_name}, {"$set": {"todo": todo}})
+        dict = {
+            'username': user_name,
```
Comment (sahare92) [inline user_management.py]:
```suggestion
            'ownerId': user_name,

```

File: user_management.py [hunk 7]
```diff
@@ -85,19 +86,30 @@ def list_users(self) -> dict:
...
+        }
+        self.todo_collection.insert_one(dict)
         return {}
 
     def get_todo_list(self,):
-        all_list = []
-        for o in self.collection.find({}, {'username': 1,'todo': 1, '_id':0}):
-            all_list.append(o)
+        parsed_todo_list = []
+        todo_list = self.todo_collection.find({}, {'username': 1,'todo': 1, })
```
Comment (sahare92) [inline user_management.py]: make this function return todos according to a given userId
meaning, this function will receive a userId and will return all todos of that user only!

File: user_management.py [hunk 8]
```diff
@@ -85,19 +86,30 @@ def list_users(self) -> dict:
...
+        for todo in todo_list:
+            print(todo)
+            parsed_todo_list.append({
+                '_id': str(todo['_id']),
+                'username': todo['username'],
+                'todo': todo['todo']
+            })
         all_todos = {
-            'todos': all_list
+            'todos': parsed_todo_list
```
Comment (sahare92) [inline user_management.py]: again, change to 'data'

**Review by sahare92 — CHANGES_REQUESTED** [review]
Very good!
Remove weird files added by accident (like `__pycache__/user_management.cpython-38.pyc`)
and a few more comments here

**Review by sahare92 — APPROVED** [review]

**Pull request merged**